Dockerfile

```diff
@@ -48,69 +52,63 @@ ARG CONFIG="--prefix=/etc/nginx \
  --with-compat \
  --add-dynamic-module=/usr/src/ngx_brotli"
 
-FROM debian AS build
-
-ARG NGINX_VERSION
-ARG NGINX_COMMIT
-ARG NGX_BROTLI_COMMIT
-ARG QUICTLS_COMMIT
-ARG CONFIG
-
 # Install dependencies
+# we need an up-to-date cargo
 RUN apt-get update && apt-get install -y \
-    dpkg-dev \
-    build-essential \
-    mercurial \
-    gnupg2 \
-    git \
-    gcc \
-    cmake \
-    libpcre3 libpcre3-dev \
-    zlib1g zlib1g-dev \
-    openssl \
-    libssl-dev \
-    curl \
-    unzip \
-    wget \
-    libxslt-dev \
-  && rm -rf /var/lib/apt/lists/*
+  dpkg-dev \
+  build-essential \
+  mercurial \
+  gnupg2 \
+  git \
+  gcc \
+  cmake \
+  libpcre3 libpcre3-dev \
+  zlib1g zlib1g-dev \
+  openssl \
+  libssl-dev \
+  curl \
+  unzip \
+  wget \
+  libxslt-dev \
+ && rm -rf /var/lib/apt/lists/* \
+ && curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
 
 WORKDIR /usr/src
 
 RUN echo "Cloning brotli $NGX_BROTLI_COMMIT" \
-  && mkdir /usr/src/ngx_brotli \
-  && cd /usr/src/ngx_brotli \
-  && git init \
-  && git remote add origin https://github.com/google/ngx_brotli.git \
-  && git fetch --depth 1 origin $NGX_BROTLI_COMMIT \
-  && git checkout --recurse-submodules -q FETCH_HEAD \
-  && git submodule update --init --depth 1
-
-RUN echo "Cloning and building quictls $QUICTLS_COMMIT" \
-  && cd /usr/src \
-  && git clone https://github.com/quictls/openssl \
-  && cd openssl \
-  && git checkout $QUICTLS_COMMIT \
-  && ./config shared --prefix=/usr/src/openssl/build --openssldir=/usr/src/openssl/build --libdir=lib \
-  && make \
-  && make install
-
-RUN echo "Cloning nginx and building $NGINX_VERSION (rev $NGINX_COMMIT from 'quic' branch)" \
-  && hg clone -b quic --rev $NGINX_COMMIT https://hg.nginx.org/nginx-quic /usr/src/nginx-$NGINX_VERSION \
-  && cd /usr/src/nginx-$NGINX_VERSION \
-  && ./auto/configure $CONFIG \
-    --with-cc-opt="-I../openssl/build/include" \
-    --with-ld-opt="-L../openssl/build/lib" \
-  && make \
-  && make install
+ && mkdir /usr/src/ngx_brotli \
+ && cd /usr/src/ngx_brotli \
+ && git init \
+ && git remote add origin https://github.com/google/ngx_brotli.git \
+ && git fetch --depth 1 origin $NGX_BROTLI_COMMIT \
+ && git checkout --recurse-submodules -q FETCH_HEAD \
+ && git submodule update --init --depth 1
+
+# using boringssl here
+RUN echo "Cloning and getting the quiche patches $QUICHE_COMMIT" \
+ && git clone --recursive https://github.com/cloudflare/quiche \
+ && cd /usr/src/quiche \
+ && git checkout --recurse-submodules $QUICHE_COMMIT \
+ && cd /usr/src \
+ && wget -q https://raw.githubusercontent.com/kn007/patch/1062e64ead7e1b21a52392cdd02d1d5bc631d231/nginx_with_quic.patch \
+ && wget -q https://raw.githubusercontent.com/kn007/patch/cd03b77647c9bf7179acac0125151a0fbb4ac7c8/Enable_BoringSSL_OCSP.patch
```

Comment:
What's this patch for if stapling will be disabled?

Reply:
I added it in case wanted to enable the stapling file. We can remove it if we don't think we'll use that feature

Reply:
Gotcha, let's move ahead and see if performance improves. If it does, we can figure out a way to do the manual stapling file